api/v1alpha1/deployment_types.go

```diff
@@ -0,0 +1,73 @@
+package v1alpha1
+
+import (
+	apps "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	// DeploymentStrategyAnnotation is annotation for deployment,
+	// which is strategy fields of Advanced Deployment.
+	DeploymentStrategyAnnotation = "rollouts.kruise.io/deployment-strategy"
+
+	// DeploymentExtraStatusAnnotation is annotation for deployment,
+	// which is extra status field of Advanced Deployment.
+	DeploymentExtraStatusAnnotation = "rollouts.kruise.io/deployment-extra-status"
+
+	// DeploymentRolloutPolicyAnnotation must be "batch" or "canary":
+	// * "partitioned" means rolling in batches just like CloneSet, and will NOT create any extra Deployment;
+	// * "canary" means rolling in canary way, and will create a canary Deployment.
+	// Defaults to canary
+	DeploymentRolloutPolicyAnnotation = "rollouts.kruise.io/deployment-rolling-policy"
+)
+
+// DeploymentStrategy is strategy field for Advanced Deployment
+type DeploymentStrategy struct {
+	// original deployment strategy fields
+	apps.DeploymentStrategy
```

Comment:
is it necessary to duplicate apps.DeploymentStrategy here ? can we use the deploymentStrategy in the deployment spec?